launcher/ui/pages/instance/ModFolderPage.cpp

```diff
@@ -145,10 +146,17 @@ void ModFolderPage::downloadMods()
         QMessageBox::critical(this, tr("Error"), tr("Please install a mod loader first!"));
         return;
     }
-    auto mdownload = new ResourceDownload::ModDownloadDialog(this, m_model, m_instance);
-    mdownload->setAttribute(Qt::WA_DeleteOnClose);
-    connect(this, &QObject::destroyed, mdownload, &QDialog::close);
-    if (mdownload->exec()) {
+    
+    m_downloadDialog = new ResourceDownload::ModDownloadDialog(this, m_model, m_instance);
```

Comment:
this is not the only place where we create a download dialog in this file:
check changeModVersion (this is also for other resources)

Reply:
Thanks for pointing that out